src/htmx.js

```diff
@@ -1091,19 +1095,19 @@ return (function () {
                         return;
                     }
 
-                    ///////////////////////////
-                    // TODO: merge this code with AJAX and WebSockets code in the future.
-
-                    var response = event.data;
-                    withExtensions(elt, function(extension){
-                        response = extension.transformResponse(response, null, elt);
-                    });
-
-                    var swapSpec = getSwapSpecification(elt)
-                    var target = getTarget(elt)
-                    var settleInfo = makeSettleInfo(elt);
+                    if (doSwap) {
+                        ///////////////////////////
+                        // TODO: merge this code with AJAX and WebSockets code in the future.
+                        var response = event.data;
+                        withExtensions(elt, function(extension){
+                            response = extension.transformResponse(response, null, elt);
+                        });
+                        var swapSpec = getSwapSpecification(elt)
+                        var target = getTarget(elt)
+                        var settleInfo = makeSettleInfo(elt);
+                        selectAndSwap(swapSpec.swapStyle, elt, target, response, settleInfo)
+                    }
 
-                    selectAndSwap(swapSpec.swapStyle, elt, target, response, settleInfo)
                     triggerEvent(elt, "htmx:sseMessage", event)
```

Comment:
I actually think we need a different name for each event, so I'm changing my mind on this. Going to try some experiments first.

Reply:
```js
                    triggerEvent(elt, "htmx:sse:" + event.name, event)
```
?

Reply:
I'm trying something like that ... but I need to do some more investigation first.

Reply:
I'm jumping in late here, but wouldn't this make it more difficult to listen to different kinds of sse messages?  If there's just one event name, then I'd only need one function, which could just switch on the message name to differentiate.  Maybe it's an issue of architecture.